// @vue-macros/common@0.12.0/dist/chunk-AKOGBVWL.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});// src/constants.ts
var DEFINE_PROPS = "defineProps";
var DEFINE_EMITS = "defineEmits";
var DEFINE_OPTIONS = "defineOptions";
var DEFINE_MODEL = "defineModel";
var DEFINE_SETUP_COMPONENT = "defineSetupComponent";
var DEFINE_RENDER = "defineRender";
var REPO_ISSUE_URL = "https://github.com/sxzz/unplugin-vue-macros/issues";
var MAGIC_COMMENT_STATIC = "hoist-static";
var REGEX_SRC_FILE = /\.[cm]?[jt]sx?$/;
var REGEX_TS_FILE = /^[cm]?tsx?$/;
var REGEX_JSX_FILE = /^[cm]?[jt]sx$/;
var REGEX_SETUP_SFC = /\.setup\.[cm]?[jt]sx?$/;
var REGEX_VUE_SFC = /\.vue$/;















exports.DEFINE_PROPS = DEFINE_PROPS; exports.DEFINE_EMITS = DEFINE_EMITS; exports.DEFINE_OPTIONS = DEFINE_OPTIONS; exports.DEFINE_MODEL = DEFINE_MODEL; exports.DEFINE_SETUP_COMPONENT = DEFINE_SETUP_COMPONENT; exports.DEFINE_RENDER = DEFINE_RENDER; exports.REPO_ISSUE_URL = REPO_ISSUE_URL; exports.MAGIC_COMMENT_STATIC = MAGIC_COMMENT_STATIC; exports.REGEX_SRC_FILE = REGEX_SRC_FILE; exports.REGEX_TS_FILE = REGEX_TS_FILE; exports.REGEX_JSX_FILE = REGEX_JSX_FILE; exports.REGEX_SETUP_SFC = REGEX_SETUP_SFC; exports.REGEX_VUE_SFC = REGEX_VUE_SFC;
