// @vue-macros/common@0.12.0/dist/index.mjs
import {
  babelParse,
  checkInvalidScopeReference,
  getLang,
  getStaticKey,
  isCallOf,
  isLiteralType,
  isStaticExpression,
  isTs,
  walkAST
} from "./chunk-ECMS2VIW.mjs";
import {
  DEFINE_EMITS,
  DEFINE_MODEL,
  DEFINE_OPTIONS,
  DEFINE_PROPS,
  DEFINE_RENDER,
  DEFINE_SETUP_COMPONENT,
  MAGIC_COMMENT_STATIC,
  REGEX_JSX_FILE,
  REGEX_SETUP_SFC,
  REGEX_SRC_FILE,
  REGEX_TS_FILE,
  REGEX_VUE_SFC,
  REPO_ISSUE_URL
} from "./chunk-2CWCRQZP.mjs";
import {
  MagicString
} from "./chunk-C7CZZFOH.mjs";
import {
  getTransformResult
} from "./chunk-JETQRGTE.mjs";
import {
  addNormalScript,
  parseSFC
} from "./chunk-GC4CLE2C.mjs";
export {
  DEFINE_EMITS,
  DEFINE_MODEL,
  DEFINE_OPTIONS,
  DEFINE_PROPS,
  DEFINE_RENDER,
  DEFINE_SETUP_COMPONENT,
  MAGIC_COMMENT_STATIC,
  MagicString,
  REGEX_JSX_FILE,
  REGEX_SETUP_SFC,
  REGEX_SRC_FILE,
  REGEX_TS_FILE,
  REGEX_VUE_SFC,
  REPO_ISSUE_URL,
  addNormalScript,
  babelParse,
  checkInvalidScopeReference,
  getLang,
  getStaticKey,
  getTransformResult,
  isCallOf,
  isLiteralType,
  isStaticExpression,
  isTs,
  parseSFC,
  walkAST
};
