// @vue-macros/common@0.12.0/dist/index.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});









var _chunkJKQLS7G6js = require('./chunk-JKQLS7G6.js');














var _chunkAKOGBVWLjs = require('./chunk-AKOGBVWL.js');


var _chunkM6KSF6PGjs = require('./chunk-M6KSF6PG.js');


var _chunk7LA3R3RZjs = require('./chunk-7LA3R3RZ.js');



var _chunkREKFGCBZjs = require('./chunk-REKFGCBZ.js');



























exports.DEFINE_EMITS = _chunkAKOGBVWLjs.DEFINE_EMITS; exports.DEFINE_MODEL = _chunkAKOGBVWLjs.DEFINE_MODEL; exports.DEFINE_OPTIONS = _chunkAKOGBVWLjs.DEFINE_OPTIONS; exports.DEFINE_PROPS = _chunkAKOGBVWLjs.DEFINE_PROPS; exports.DEFINE_RENDER = _chunkAKOGBVWLjs.DEFINE_RENDER; exports.DEFINE_SETUP_COMPONENT = _chunkAKOGBVWLjs.DEFINE_SETUP_COMPONENT; exports.MAGIC_COMMENT_STATIC = _chunkAKOGBVWLjs.MAGIC_COMMENT_STATIC; exports.MagicString = _chunkM6KSF6PGjs.MagicString; exports.REGEX_JSX_FILE = _chunkAKOGBVWLjs.REGEX_JSX_FILE; exports.REGEX_SETUP_SFC = _chunkAKOGBVWLjs.REGEX_SETUP_SFC; exports.REGEX_SRC_FILE = _chunkAKOGBVWLjs.REGEX_SRC_FILE; exports.REGEX_TS_FILE = _chunkAKOGBVWLjs.REGEX_TS_FILE; exports.REGEX_VUE_SFC = _chunkAKOGBVWLjs.REGEX_VUE_SFC; exports.REPO_ISSUE_URL = _chunkAKOGBVWLjs.REPO_ISSUE_URL; exports.addNormalScript = _chunkREKFGCBZjs.addNormalScript; exports.babelParse = _chunkJKQLS7G6js.babelParse; exports.checkInvalidScopeReference = _chunkJKQLS7G6js.checkInvalidScopeReference; exports.getLang = _chunkJKQLS7G6js.getLang; exports.getStaticKey = _chunkJKQLS7G6js.getStaticKey; exports.getTransformResult = _chunk7LA3R3RZjs.getTransformResult; exports.isCallOf = _chunkJKQLS7G6js.isCallOf; exports.isLiteralType = _chunkJKQLS7G6js.isLiteralType; exports.isStaticExpression = _chunkJKQLS7G6js.isStaticExpression; exports.isTs = _chunkJKQLS7G6js.isTs; exports.parseSFC = _chunkREKFGCBZjs.parseSFC; exports.walkAST = _chunkJKQLS7G6js.walkAST;
